src/ImageSharp/Formats/Tga/TgaEncoderCore.cs

```diff
@@ -225,35 +228,29 @@ private void WriteRunLengthEncodedImage<TPixel>(Stream stream, ImageFrame<TPixel
         /// <typeparam name="TPixel">The pixel type.</typeparam>
         /// <param name="pixels">The pixels of the image.</param>
         /// <param name="xStart">X coordinate to start searching for the same pixels.</param>
-        /// <param name="yStart">Y coordinate to start searching for the same pixels.</param>
+        /// <param name="yPos">Y coordinate to searching for the same pixels in only one scan line.</param>
         /// <returns>The number of equal pixels.</returns>
-        private byte FindEqualPixels<TPixel>(Buffer2D<TPixel> pixels, int xStart, int yStart)
+        private byte FindEqualPixels<TPixel>(Buffer2D<TPixel> pixels, int xStart, int yPos)
             where TPixel : unmanaged, IPixel<TPixel>
         {
             byte equalPixelCount = 0;
-            bool firstRow = true;
-            TPixel startPixel = pixels[xStart, yStart];
-            for (int y = yStart; y < pixels.Height; y++)
+            TPixel startPixel = pixels[xStart, yPos];
```

Comment:
Instead of using the indexer, you could use `DangerousGetRowSpan`, which should be faster. Like this:

```
Span<TPixel> pixelRow = pixels.DangerousGetRowSpan(yPos).Slice(xStart);
```